aroma/utils.py

```diff
@@ -275,73 +74,6 @@ def cross_correlation(a, b):
     return np.corrcoef(a.T, b.T)[:ncols_a, ncols_a:]
 
 
-def classification(features_df, out_dir):
-    """Classify components as motion or non-motion based on four features.
-
-    The four features used for classification are: maximum RP correlation,
-    high-frequency content, edge-fraction, and CSF-fraction.
-
-    Parameters
-    ----------
-    features_df : (C x 4) pandas.DataFrame
-        DataFrame with the following columns:
-        "edge_fract", "csf_fract", "max_RP_corr", and "HFC".
-    out_dir : str
-        Full path of the output directory
-
-    Returns
-    -------
-    motion_ICs : array_like
-        Array containing the indices of the components identified as motion
-        components
-
-    Output
-    ------
-    classified_motion_ICs.txt : A text file containing the indices of the
-                                components identified as motion components
-    """
-    # Put the feature scores in a text file
-    features_df.to_csv(op.join(out_dir, "feature_scores.tsv"), sep="\t", index=False)
-
-    # Classify the ICs as motion or non-motion
-
-    # Define criteria needed for classification (thresholds and
-    # hyperplane-parameters)
-    THR_CSF = 0.10
-    THR_HFC = 0.35
-    HYPERPLANE = np.array([-19.9751070082159, 9.95127547670627, 24.8333160239175])
-
-    # Project edge & max_RP_corr feature scores to new 1D space
-    x = features_df[["max_RP_corr", "edge_fract"]].values
-    proj = HYPERPLANE[0] + np.dot(x.T, HYPERPLANE[1:])
-
-    # Classify the ICs
-    is_motion = (
-        (features_df["csf_fract"] > THR_CSF)
-        | (features_df["HFC"] > THR_HFC)
-        | (proj > 0)
-    )
-    features_df["classification"] = is_motion
-    features_df["classification"] = features_df["classification"].map(
-        {True: "rejected", False: "accepted"}
-    )
-
-    # Put the indices of motion-classified ICs in a text file (starting with 1)
-    motion_ICs = features_df[
-        "classification", features_df["classification"] == "rejected"
-    ].index.values
-    with open(op.join(out_dir, "classified_motion_ICs.txt"), "w") as fo:
-        out_str = ",".join(motion_ICs.astype(str))
-        fo.write(out_str)
-
-    # Create a summary overview of the classification
-    features_df.to_csv(
-        op.join(out_dir, "classification_overview.txt"), sep="\t", index_label="IC"
-    )
```

Comment:
Since this file is no longer written out here, it would be good to write it out in the workflow. We also need a corresponding change in the workflow function, I think.